guard/src/migrate/parser.rs

```diff
@@ -180,10 +192,9 @@ impl Display for RuleLineType {
 
 // variable-dereference =  ("%" variable) / ("%{" variable "}" ); Regular and environment variables, respectively
 pub (crate) fn parse_variable_dereference(input: Span) -> IResult<Span, String> {
-    preceded(space0, alt((
+    delimited(space0, alt((
```

Comment:
Please add an explanation on why this change was needed for the next set of maintainers. It is hard to understand why this change was made without the context surrounding this

Reply:
I get that gist from the change made in the code :). Elaborate on the context for 
1. Why was stripping white spaces needed? 
2. What was the error/consequence when this wasn't done?